tests/onlinequantile/main.c

```diff
@@ -0,0 +1,173 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "onlinequantile.h"
+#include "parsenum.h"
+#include "warnp.h"
+
+#define LINELEN 80
+
+/* Return 0 if ok, 1 on EOF, -1 on error. */
+static int
+parse_line(FILE * fp, char * instr, double * val)
+{
+	char line[LINELEN];
+
+	/* Repeat until we have a non-comment, non-blank line. */
+	do {
+		if (fgets(line, LINELEN, fp) == NULL) {
+			/* Handle error or EOF. */
+			if (ferror(fp)) {
+				warnp("fgets");
+				goto err0;
+			}
+			goto eof;
+		}
+	} while ((line[0] == '#') || (strlen(line) == 1));
+
+	/* Sanity-check the format.  Multiple tabs characters are ok. */
+	if ((line[1] != '\t') || (line[strlen(line) - 1] != '\n')) {
+		warn0("Unexpected character in line: %s", line);
+		goto err0;
+	}
+
+	/* Remove newline. */
+	line[strlen(line) - 1] = '\0';
```

Comment:
The code is cleaner if you put this into the `do { } while` loop above; that way "empty line" becomes the far more intuitive "line of length zero".